mininet/node.py

```diff
@@ -848,27 +848,46 @@ def __init__( self, name, **kwargs ):
 
     def config( self, ad=None, hid=None, lpm=None,
                 u4id=None, zfid=None, xdp=False, **params ):
-        """hid: private key file names for Host Identifiers.
-           zfid: zFilter identifier.
-           xdp: set True to load the xdp module
+        """ad: list of Autonomous Domain IDs.
+           hid: list of private key file names for Host Identifiers.
+           lpm: list of lpm inputs in lpmid/prefixLen form.
+           u4id: dictionary of inputs for tunneling.
+           xdp: set True to load the xdp module.
+           zfid: list of zFilter identifiers.
            params: parameters for Node.config()"""
         r = Node.config( self, **params )
         if xdp is True:
             moduleDeps( add='xia_ppal_xdp' )
         self.setParam( r, 'setAD', ad=ad )
         self.setParam( r, 'setHID', hid=hid )
-        self.setParam( r, 'setLPM' , lpm=lpm )
+        self.setParam( r, 'setLPM', lpm=lpm )
         self.setParam( r, 'setU4ID', u4id=u4id )
         self.setParam( r, 'setZFID', zfid=zfid )
         return r
 
+    def validateXID( self, xid, base, length=-1 ):
```

Comment:
The base should always be 16 (hexadecimal). Even in the case of zFilter IDs, we sometimes write them as just strings of 0s and 1s, but each one of these 0s and 1s is a nibble, not a bit. For example, a zFilter ID that starts with 0xF... is really 1111... etc.

So we can keep this check in here, but we don't need to pass in the base as a parameter.

For the length, let's use 40 as a default, and pass -1 in the case of LPM (since that is the only exception we have). Let's not use this function to test for valid port numbers, since it should be for validating XIDs only.